src/python_testing/TC_OCC_2_1.py

```diff
@@ -0,0 +1,253 @@
+            asserts.assert_less(occupancy_sensor_type_dut, Clusters.Objects.OccupancySensing.Enums.OccupancySensorTypeEnum.kUnknownEnumValue,
+                                "OccupancySensorType is not in valid range")
+        else:
+            logging.info("OccupancySensorType attribute is a mandatory attribute. Test step fails")
+            asserts.fail("Missing mandatory attribute OccupancySensorType")
+
+        self.step(4)
+        if attributes.OccupancySensorTypeBitmap.attribute_id in attribute_list:
+            occupancy_sensor_type_bitmap_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.OccupancySensorTypeBitmap)
+            asserts.assert_less_equal(occupancy_sensor_type_bitmap_dut, 0b00000111, "OccupancySensorTypeBitmap attribute is not in valid range")
+        else:
+            logging.info("OccupancySensorTypeBitmap attribute is a mandatory attribute. Test step fails")
+            asserts.fail("Missing mandatory attribute OccupancySensorTypeBitmap")
+
+        self.step(5)
+        if attributes.HoldTimeLimits.attribute_id in attribute_list:
+            if attributes.HoldTime.attribute_id in attribute_list: # check HoldTime conformance
+                hold_time_limits_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.HoldTimeLimits)
+
+                asserts.assert_less_equal(hold_time_limits_dut.HoldTimeMin, hold_time_limits_dut.HoldTimeMax, "HoldTimeMin is not in valid range")
+                asserts.assert_greater_equal(hold_time_limits_dut.HoldTimeMin, 0, "HoldTimeMin is not in valid range")
+                asserts.assert_less_equal(hold_time_limits_dut.HoldTimeMax, 0xFFFE, "HoldTimeMin is not in valid range")
+                asserts.assert_greater_equal(hold_time_limits_dut.HoldTimeMax, hold_time_limits_dut.HoldTimeMin, "HoldTimeMin is not in valid range")
+                asserts.assert_less_equal(hold_time_limits_dut.HoldTimeDefault, hold_time_limits_dut.HoldTimeMax, "HoldTimeMin is not in valid range")
+                asserts.assert_greater_equal(hold_time_limits_dut.HoldTimeDefault, hold_time_limits_dut.HoldTimeMin, "HoldTimeMin is not in valid range")
+            else:
+                logging.info("HoldTime conformance failed.  Test step skipped")
+                asserts.fail("HoldTime conformance is incorrect")
+
+        else:
+            logging.info("HoldTimeLimits not supported. Test step skipped")
+            self.mark_current_step_skipped()
+        
+        self.step(6)
+        if attributes.HoldTime.attribute_id in attribute_list:
+            hold_time_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.HoldTime)
+
+            asserts.assert_less_equal(hold_time_dut, 0xFFFE, "HoldTime attribute is out of range")
+            asserts.assert_greater_equal(hold_time_dut, 0, "HoldTime attribute is out of range")

```

Comment:
This needs to validate against HoldTimeLimits if they exist according to the test plan:

```Verify that its value is greater than equal to HoldTimeMin and less than equal to HoldTimeMax of OCC.S.A0004(HoldTimeLimits) data struct```

Reply:
Done